File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.java
import java.math.BigInteger;

class Solution {
    public int numSteps(String s) {
        BigInteger n = new BigInteger(s, 2);
        int c = 0;
        
        while (!n.equals(BigInteger.ONE)) {
            if (n.mod(BigInteger.TWO).equals(BigInteger.ZERO)) {
                n = n.divide(BigInteger.TWO);
            } else {
                n = n.add(BigInteger.ONE);
            }
            c++;
        }
        
        return c;
    }
}

File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
    def numSteps(self, s: str) -> int:
        n = int(s, 2)  
        c = 0
        
        while n != 1:
            if n % 2 == 0:
                n //= 2
            else:
                n += 1
            c += 1
        
        return c
